modules/virt-creating-configmap.adoc

```diff
@@ -0,0 +1,107 @@
+// Module included in the following assemblies:
+// * virt/virtual_machines/importing_vms/virt-importing-rhv-vm.adoc
+[id='virt-creating-configmap_{context}']
+= Creating a custom operating system ConfigMap
+
+If the operating system of the Red Hat Virtualization virtual machine does not exist in the default ConfigMap of the `vm-import-controller` Custom Resource (CR), you can create a custom ConfigMap to map the operating system to a common template. The template is used to create the new virtual machine in {VirtProductName}.
+
+.Procedure
+
+. Create a file named `configmap.txt` that maps the RHV virtual machine's operating system to a common template, as in the following example:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: os_configmap
+  namespace: default <1>
+data:
+  guestos2common: | <2>
+    "Red Hat Enterprise Linux Server": "rhel"
+    "CentOS Linux": "centos"
+    "Fedora": "fedora"
+    "Ubuntu": "ubuntu"
+    "openSUSE": "opensuse"
+  osinfo2common: | <3>
+    "rhel_6_9_plus_ppc64": "rhel6.9"
+    "rhel_6_ppc64": "rhel6.9"
+    "rhel_6": "rhel6.9"
+    "rhel_6x64": "rhel6.9"
+    "rhel_7_ppc64": "rhel7.7"
+    "rhel_7_s390x": "rhel7.7"
+    "rhel_7x64": "rhel7.7"
+    "rhel_8x64": "rhel8.1"
+    "sles_11_ppc64": "opensuse15.0"
+    "sles_11": "opensuse15.0"
+    "sles_12_s390x": "opensuse15.0"
+    "ubuntu_12_04": "ubuntu18.04"
+    "ubuntu_12_10": "ubuntu18.04"
+    "ubuntu_13_04": "ubuntu18.04"
+    "ubuntu_13_10": "ubuntu18.04"
+    "ubuntu_14_04_ppc64": "ubuntu18.04"
+    "ubuntu_14_04": "ubuntu18.04"
+    "ubuntu_16_04_s390x": "ubuntu18.04"
+    "windows_10": "win10"
+    "windows_10x64": "win10"
+    "windows_2003": "win10"
+    "windows_2003x64": "win10"
+    "windows_2008R2x64": "win2k8"
+    "windows_2008": "win2k8"
+    "windows_2008x64": "win2k8"
+    "windows_2012R2x64": "win2k12r2"
+    "windows_2012x64": "win2k12r2"
+    "windows_2016x64": "win2k16"
+    "windows_2019x64": "win2k19"
+    "windows_7": "win10"
+    "windows_7x64": "win10"
+    "windows_8": "win10"
+    "windows_8x64": "win10"
+    "windows_xp": "win10"
+----
+<1> Optional: you can change the namespace.
+<2> Specify the operating system. `guestos2common` maps the operating system to a common template.
+<3> Specify the operating system version. `osinfo2common` maps the operating system resource of the source provider to a common template.
+
+. Create a custom ConfigMap:
++
+----
+$ oc create configmap os_configmap --from-file=./configmap_file.txt
+----
+
+. Update the `vm-import-operator` deployment:
++
+----
+$ oc patch deployment vm-import-operator -n default --patch '
```

Comment:
@apinnick there could be a difference between the upstream to downstream process for updating the vm-import-operator deployment. Once I'll get an access to d/s env, I'll be able to verify this and we'll write the required steps for setting the custom OS config map values.

Reply:
@apinnick I verified the process of updating setting the OS config map values is different for the d/s, hence this section needs to be changed according to this:

1. from the UI, there is a need to pick the Openshift Virtualization operator:
![Selection_999(897)](https://user-images.githubusercontent.com/316242/85790427-de121500-b738-11ea-8ae9-7964e94560f6.png)
2. Update the YAML definition of kubevirt-hyperconverged-operator

or from the CLI use (assuming CNV version is 2.4.0):
`oc edit clusterserviceversion -n openshift-cnv kubevirt-hyperconverged-operator.v2.4.0`
And Update the YAML definition.

In both cases (UI and CLI), the YAML should be changed as follow:
Find the deployment named 'vm-import-operator' (under spec --> install --> deployments)
and set the values of the elements under the container's env:
```
                - name: OS_CONFIGMAP_NAMESPACE
                  value: default
                - name: OS_CONFIGMAP_NAME
                  value: os-configmap
```
Here's a snippet of the YAML with the values for the OS config map:
![Selection_999(899)](https://user-images.githubusercontent.com/316242/85793524-d012c300-b73d-11ea-9376-728638a953ad.png)

The step describe in this comment replaces lines 68-107.
No need to perform any other actions once the YAML is saved.
As a result of changing the HCO  clusterserviceversion resource, the vm-import-operator deployment will be updated which in its turn updates the vm-import-controller.
 
As a side-note:
This method of updating the CSV resource is designed for cnv-2.4 only.
We'll implement less intrusive method to update the deployment resource of the vm-import-controller in cnv-2.5, where setting the OS config map values will be executed by updating the VmImportConfig. This will be tracked by a 2.5 bug.

Reply:
> 2. Update the YAML definition of kubevirt-hyperconverged-operator

Is the downstream operator called `vm-import-operator` or `kubevirt-hyperconverged-operator`?